### PR TITLE
Tests for the new UX (re: #910)

### PR DESCRIPTION
**Pull request by baltpeter**

### CODE REVIEW

**Comment by baltpeter** [comment]
Remaining:

* The tests sometimes fail to run in CI. Cypress complains that it can't reach the dev server.
* The command for the generator store is flaky for some reason (but the one for the proceedings store seems to work fine).
* The [object Object] test will fail until the letter-generator PR is merged and the package updated here.

**Comment by baltpeter** [comment]
> The tests sometimes fail to run in CI. Cypress complains that it can't reach the dev server.

This is also happening on master and I'm really not sure what the problem is. It's _not_ a timeout problem as we've had multiple times in the past. The first n tests do pass, so Hugo must have finished building successfully. But then at some point (which seems to vary between runs), Cypress can't connect to the dev server anymore. Maybe Hugo crashes for some reason?

I've not been able to reproduce the problem locally.

**Comment by baltpeter** [comment]
Couldn't even reproduce it in Circle's SSH mode. :/

**Comment by baltpeter** [comment]
> The command for the generator store is flaky for some reason (but the one for the proceedings store seems to work fine).

Ok, the problem is actually quite obvious. The property is only set in a `useEffect`: https://github.com/datenanfragen/website/blob/44984086b094c8079465dc4e7179d02ddb5b97ca/src/app.tsx#L11-L16

It isn't exactly surprising that that is flaky. 

I guess, I'll have to implement a wait mechanism for that… *shrug*

**Comment by baltpeter** [comment]
I've tried using a higher resource class in Circle to hopefully avoid the Hugo problem. We'll have to see whether that actually works.

That still leaves the context problem, though…

**Comment by zner0L** [comment]
I was thinking that maybe using the cypress wait function like `cypress-wait-until` does instead of a timeout might help with the wait issues. But I don't want to include the packages because it is the typical of [a one file npm package](https://github.com/NoriSte/cypress-wait-until/blob/master/src/index.js). At least no dependencies. What do you think, @baltpeter?

**Comment by zner0L** [comment]
I rebased and hopefully fixed the context problem. I would like you to look at my commit since this is your PR, and if it checks out you can merge, @baltpeter.

**Review by zner0L — APPROVED** [review]

File: src/Components/Generator/GeneratorTestInterface.tsx
Comment (baltpeter) [inline src/Components/Generator/GeneratorTestInterface.tsx:24]: That seems overly complicated. You're assigning anyway. Why not just…?

```ts
(window as WindowWithGeneratorApi).generatorStoreApi[props.generatorId || 'default'] = generatorStoreApi;
```

**Review by baltpeter — COMMENTED** [review]
Did you make changes in commits other than "Use waitUntil to wait…"? If so, I didn't see them due to the rebase changes.

See below for my comments. Other than that, it's fine to merge if the tests pass now.

File: src/Components/Generator/GeneratorTestInterface.tsx
Comment (zner0L) [reply src/Components/Generator/GeneratorTestInterface.tsx:24]: That threw an error if `window.generatorStoreApi` is undefined.

File: src/Components/Generator/GeneratorTestInterface.tsx
Comment (zner0L) [reply src/Components/Generator/GeneratorTestInterface.tsx:26]: That was an oversight.

**Pull request merged**
